### PR TITLE
Fixing up build.args etc for import and deploy

### PR DESCRIPTION
**Pull request by codepope**

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -144,6 +144,13 @@ func runDeploy(commandContext *cmdctx.CmdContext) error {
 	} else {
 		// no image specified, build one
 		buildArgs := map[string]string{}
+
+		if commandContext.AppConfig.Build.Args != nil {
```
Comment (michaeldwan) [inline cmd/deploy.go:148]: You might need to check that `commandContext.AppConfig.Build` isn't nil first. Try it out with an app without a `[build]` section

File: flyctl/app_config.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ const (
 type AppConfig struct {
 	AppName string
 	Build   *Build
-
+	// DockerBuildArgs map[string]string
```
Comment (michaeldwan) [inline flyctl/app_config.go:28]: just nuke it 😉

**Review by michaeldwan — CHANGES_REQUESTED** [review]
Looks good, but I _think_ you'll need a nil check

File: cmd/deploy.go (hunk 1 above)
Comment (codepope) [reply cmd/deploy.go:148]: Ran it without a [build] and it worked, will add in the check tho. I suspect that the toml library fills in for ancestors... ie, build.args infers build as a section. So if there's a build.x, there will be an build to hold the .x in the tree. Our marshall to Toml code ends up putting in a build too which gets added to the .toml file.

So I think it's ok.

File: flyctl/app_config.go (hunk 2 above)
Comment (codepope) [reply flyctl/app_config.go]: Oh dash, nuked now. Must get into the habit of labelling kipple comments.

File: cmd/deploy.go (hunk 1 above)
Comment (codepope) [reply cmd/deploy.go:148]: Well, I was so wrong there. Blows up when there's no build or build subsections at all. So... fix incoming.

**Pull request merged**